yeonjae/baekjoon/큐/11866-요세푸스문제.js

```diff
@@ -0,0 +1,31 @@
+const readline = require("readline");
+const rl = readline.createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const input = [];
+
+rl.on("line", function (line) {
+  input.push(line);
+}).on("close", function () {
+  // 풀이
+  const [n, k] = input[0].split(" ").map(Number);
+  const queue = Array.from({ length: n }, (v, i) => i + 1);
+  const result = [];
+
+  while (queue.length) {
+    for (let i = 0; i < k - 1; i++) {
+      const removePerson = queue.shift();
+      queue.push(removePerson);
+    }
+    const removePerson = queue.shift();
+    result.push(removePerson);
+  }
+
+  console.log(`<${result.join(", ")}>`);
```

Comment:
```suggestion
  const [n, k] = input[0].split(" ").map(Number);
  const queue = Array.from({ length: n }, (v, i) => i + 1);
  const result = [];

  while (queue.length) {
    for (let i = 0; i < k - 1; i++) {
      const removedPerson = queue.shift();
      queue.push(removedPerson);
    }
    const removedPerson = queue.shift();
    result.push(removedPerson);
  }

  console.log(`<${result.join(", ")}>`);
```
- 풀이잘봤습니다 🙇‍♂️
- remove 라는 동사보다는 removed가 적절한 네이밍인 것 같습니다 ~
- 아래처럼 count와 조건식을 이용해 for문이 아닌 형태로 결과를 도출해낼 수 있습니다. 로직은 사실은 같습니다 :) 
```js
 const [n, k] = input[0].split(" ").map(Number);
  const queue = Array.from({ length: n }, (_, i) => i + 1);
  const result = [];
  let count = 0;

  while (queue.length) {
    const removedFirstPerson = queue.shift();

    count++;

    if (count % k === 0) {
      result.push(removedFirstPerson);
    } else {
      queue.push(removedFirstPerson);
    }
  }

  console.log(`<${result.join(", ")}>`);
```

Reply:
count를 활용해서 할수있군요! 아래 코드 참고해보겠습니다!!